KEP-3386: Graduate Evented PLEG to Beta

harche opened the pull request:
Signed-off-by: Harshal Patil <harpatil@redhat.com>

<!-- 
	Please use the following format when naming your PR
	< Issue Number >:< Issue Description >
	e.g. KEP-000: adding beta graduation criteria
	
	Avoid using phrases like `fixes #NNNN` in the description
	unless the pull request is to change the KEP status to 
	implemented or KEP has been deprecated.
-->

<!-- short description of work done in PR e.g. updating milestone, adding new KEP, adding test requirements… -->  
- One-line PR description: Graduate Evented PLEG to Beta

<!-- link to the k/enhancements issue -->
- Issue link: https://github.com/kubernetes/enhancements/issues/3386

<!-- other comments or additional information -->
- Other comments:

mrunalp commented:
/lgtm

mrunalp commented:
@derekwaynecarr ptal

mrunalp commented:
/label lead-opted-in

mrunalp commented:
/milestone v1.27

rphillips commented:
/lgtm

derekwaynecarr commented:
/approve

harche commented on keps/sig-node/3386-kubelet-evented-pleg/kep.yaml:
```diff
@@ -13,12 +13,12 @@ approvers:
+latest-milestone: "v1.27"
 
 # The milestone at which this feature was, or is targeted to be, at each stage.
 milestone:
```
Fixed it. Thanks.

ruiwen-zhao commented on keps/sig-node/3386-kubelet-evented-pleg/README.md:
```diff
@@ -341,6 +344,10 @@ We expect no non-infra related flakes in the last month as a GA graduation crite
 
+#### Beta
+- Add E2E Node Conformance presubmit job in CI 
+- Add E2E Node Conformance periodic job in CI
```
this would require containerd support I believe. Evented PLEG related support is in containerd 1.7, but I think we have a testgrid testing against main branch of containerd, which should be sufficient.

jeremyrickard reviewed (commented):
Hey there, I'm doing a first pass of some PRR reviews this cycle and I see this one hasn't been updated for beta. I think it looks pretty good, but it would be useful to include a little more context around testing rollback and upgrade. In the current PRR you mentioned `N/A for alpha release`, but as you're targeting beta now it would be nice to include some information about upgrade and rollback testing since this will be enabled. 

It would also be nice to reference any enablement/disablement unit tests that were added during the alpha work (the PRR template mentions they would be added, linking to them now assuming they were would be helpful).

harche commented:
> Hey there, I'm doing a first pass of some PRR reviews this cycle and I see this one hasn't been updated for beta. I think it looks pretty good, but it would be useful to include a little more context around testing rollback and upgrade. In the current PRR you mentioned `N/A for alpha release`, but as you're targeting beta now it would be nice to include some information about upgrade and rollback testing since this will be enabled.
> 
> It would also be nice to reference any enablement/disablement unit tests that were added during the alpha work (the PRR template mentions they would be added, linking to them now assuming they were would be helpful).
> 
> There is a new question in the scalability section. Could you add that to this PR and answer the question (see [78200cb](https://github.com/kubernetes/enhancements/commit/78200cb9d1a4244fd068e819c4cb430ebd114eaa) for the change).

Thanks for your review. I have addressed those points in latest changes.

deads2k commented:
The `What specific metrics should inform a rollback?` looks light to me.  For alpha, it was ok to say, "check this value on the host", but for beta we should have metrics to assess the health of the evented PLEG connection.  Metrics for error counts, successful connections, latency, and notifications received all seem plausible to expect and could inform cluster-admins about the health of their nodes using this feature.

harche commented:
> The `What specific metrics should inform a rollback?` looks light to me. For alpha, it was ok to say, "check this value on the host", but for beta we should have metrics to assess the health of the evented PLEG connection. Metrics for error counts, successful connections, latency, and notifications received all seem plausible to expect and could inform cluster-admins about the health of their nodes using this feature.

Thanks @deads2k. I updated the PR with metrics for evented PLEG's connection with the CRI Runtime.

deads2k commented:
> Thanks @deads2k. I updated the PR with metrics for evented PLEG's connection with the CRI Runtime.

Thanks, PRR looks good.

/approve

rphillips commented:
/lgtm
